### ISSUE TITLE
IPGlobalProperties.GetActiveTcpListeners does not return ports in use by docker on MacOS

### ISSUE THREAD

**Issue by mmisztal1980** [dotnet/corefx#38570]
I have defined a helper class to get me the 1st available port on a host, starting with a port number I provide. I use it to start docker containers from C#, when trying to avoid port conflicts.

```csharp
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.NetworkInformation;

namespace MyProject
{
    public static class PortManager
    {
        public static ushort GetAvailablePort(ushort startingPort)
        {
            var portArray = new List<int>();

            var properties = IPGlobalProperties.GetIPGlobalProperties();

            portArray.AddRange(GetListenerPorts(() => properties.GetActiveTcpConnections().Select(x => x.LocalEndPoint), startingPort));
            portArray.AddRange(GetListenerPorts(() => properties.GetActiveTcpListeners(), startingPort));
            portArray.AddRange(GetListenerPorts(() => properties.GetActiveUdpListeners(), startingPort));

            portArray.Sort();

            for (int i = startingPort; i < ushort.MaxValue; i++)
            {
                if (!portArray.Contains(i))
                {
                    return Convert.ToUInt16(i);
                }
            }

            return 0;
        }

        private static IEnumerable<int> GetListenerPorts(Func<IEnumerable<IPEndPoint>> endpointSelector, ushort startinPort)
        {
            return endpointSelector().Where(x => x.Port >= startinPort).Select(x => x.Port);
        }
    }
}
```

This code works quite fine on Windows, however I've noticed that the behavior of this code on MacOS is different, as ports in use by docker (EXPOSE {port} with `-p` flag) are ignored.

- Can you comment on this behavior?
- How would you recommend to get the port numbers in use by docker on MacOS (and potentially Linux)?

**Comment by davidsh** [comment]
@wfurt

**Comment by wfurt** [comment]
What runtime do you use? I tried that with current 3.0 and it seems to work as expected. 
I also did quick run with 2.1 and it is broken as "expected" See #32727 for more information. if that is same issue you may be able to work around the endianess. 

On broader note, there is race condition between getting open ports and trying to use them. Some corefx tests have mechanism where we try port and we have retry mechanism if that fails.

**Comment by mmisztal1980** [comment]
Hi @wfurt  I've tried that on 2.1 . What about 2.2? is it also broken?

**Comment by wfurt** [comment]
As far as I can see, the fix only went to 3.0 and was never back-ported.

**Comment by mmisztal1980** [comment]
thanks @wfurt
